lib/matplotlib/backends/backend_cairo.py

```diff
@@ -171,14 +171,18 @@ def draw_image(self, gc, x, y, im):
         # bbox - not currently used
         if _debug: print('%s.%s()' % (self.__class__.__name__, _fn_name()))
 
-        rows, cols, buf = im.color_conv (BYTE_FORMAT)
-        surface = cairo.ImageSurface.create_for_data (
-                      buf, cairo.FORMAT_ARGB32, cols, rows, cols*4)
+        if sys.byteorder == 'little':
+            im = im[:, :, (2, 1, 0, 3)]
+        else:
+            im = im[:, :, (3, 0, 1, 2)]
```

Comment:
Do we need to check that image is RGBA (I am assuming that that is what the 4 bytes that are getting swapped around are for, right)? I am also going to assume that the swaps are correct. I have never been able to remember byte order stuff correctly.